### PR TITLE
Multiprocessing Manager resources (Queue) to be freed up during task stop

### PR DESCRIPTION
**Pull request by sg893052**
Multiprocessing Manager resources (Queue) to be freed up during task stop

#### Why I did it
Fix for issue https://github.com/sonic-net/sonic-buildimage/issues/14964 

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it
Free up Multiprocessing Manager resource at task stop request
[self.mpmgr.shutdown() in task_stop]

#### How to verify it
time systemctl stop system-health.service

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

**Comment by sg893052** [comment]
@adyeung @Junchao-Mellanox @vaibhavhd  
FYI: This PR will fix the https://github.com/sonic-net/sonic-buildimage/issues/14964

File: src/system-health/health_checker/sysmonitor.py [hunk 1]
```diff
@@ -466,5 +466,22 @@ def task_worker(self):
...
         self.system_service()
 
+    def task_stop(self):
+        # Signal the process to stop
+        self.task_stopping_event.set()
+        #Clear the resources of mpmgr- Queue
+        self.mpmgr.shutdown()
+
+        # Wait for the process to exit
+        self._task_process.join(self._stop_timeout_secs)
```
Comment (Junchao-Mellanox) [inline src/system-health/health_checker/sysmonitor.py:476]: should monitor_system_bus task and monitor_statedb_table task be join here?

File: src/system-health/health_checker/sysmonitor.py [hunk 2]
```diff
@@ -466,5 +466,22 @@ def task_worker(self):
...
+
+        # If the process didn't exit, attempt to kill it
+        if self._task_process.is_alive():
+            os.kill(self._task_process.pid, signal.SIGKILL)
+
+        if self._task_process.is_alive():
```
Comment (Junchao-Mellanox) [inline src/system-health/health_checker/sysmonitor.py:482]: log something here?

Comment (sg893052) [reply src/system-health/health_checker/sysmonitor.py:484]: Added logs

File: src/system-health/health_checker/sysmonitor.py (hunk 1 above)
Comment (sg893052) [reply src/system-health/health_checker/sysmonitor.py:477]: At the end of the system_service() function, there exists a call to monitor_system_bus.task_stop() and monitor_statedb_table.task_stop(), which are responsible for shutting down their respective tasks.

**Comment by bingwang-ms** [comment]
@Junchao-Mellanox  Can you please check and review? Thanks

**Comment by vaibhavhd** [comment]
@Junchao-Mellanox please check the latest comments here. We need to take this fix in sooner.

@sg893052, please update the test description to indicate that the shutdown path is no longer hung/delayed. Please test that path if not already tested.

**Review by Junchao-Mellanox — APPROVED** [review]

**Comment by vaibhavhd** [comment]
ADO: 22090981

**Review by vaibhavhd — APPROVED** [review]

**Comment by yxieca** [comment]
@sg893052 please help create separate PR for 202205/202211 branches. Thanks!

**Comment by vaibhavhd** [comment]
> @sg893052 please help create separate PR for 202205/202211 branches. Thanks!

Hi @sg893052 , pinging you on this request. The issue is blocking 202205 warm upgrades. Please help to create cherry pick PRs.

**Comment by sg893052** [comment]
Cherry-pick PR to 202205: https://github.com/sonic-net/sonic-buildimage/pull/15662
Cherry-pick PR to 202211: https://github.com/sonic-net/sonic-buildimage/pull/15663

**Comment by sg893052** [comment]
> @sg893052 PR conflicts with 202205 branch
This PR is already merged to 202205 branch using https://github.com/sonic-net/sonic-buildimage/pull/15662

**Pull request merged**